### PR TITLE
Allow building doors on buildable_to nodes.

### PR DESCRIPTION
**Pull request by sofar**
This code never allowed placing a door on e.g. a grass
plant. The code to handle this isn't that complex. With
this code, doors can be placed on flowers and on normal
node surfaces without issues.

A second patch allows opening and closing doors (etc.)
when wielding a door item.

We also check that we're not right-clicking an entity with
a door in hand, since that would crash the server.

### CODE REVIEW

**Comment by MT-Modder** [comment]
:+1: Nice and fast.

File: mods/doors/init.lua [hunk 1]
```diff
@@ -183,8 +183,24 @@ function doors.register(name, def)
...
+				return itemstack
+			end
+
+			local node = minetest.get_node(pointed_thing.under)
+			if minetest.registered_nodes[node.name].on_rightclick then
+				return minetest.registered_nodes[node.name].on_rightclick(pointed_thing.under,
+						node, placer, itemstack)
+			end
+
+			if minetest.registered_nodes[node.name].buildable_to then
```
Comment (kilbith) [inline mods/doors/init.lua]: Please localize `minetest.registered_nodes[node.name]`. Used 3 times here.

**Comment by kilbith** [comment]
By the way, what about importing this patch ? https://github.com/minetest/minetest_game/pull/831

File: mods/doors/init.lua (hunk 1 above)
Comment (sofar) [inline mods/doors/init.lua]: yeah, absolutely

**Comment by sofar** [comment]
@kilbith absolutely, while I'm working this would be silly not to include that.

**Comment by sofar** [comment]
Last push contains the change in #831 by @tenplus1

**Comment by paramat** [comment]
Added line comments.

**Comment by sofar** [comment]
I can check ``def```, yes. It's a bit far-fetched, though ;)

**Comment by sofar** [comment]
Added checks on ```def``` where needed.

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
Blocker for release.

**Comment by Ekdohibs** [comment]
:+1:

**Comment by paramat** [comment]
Will merge soon.

**Comment by paramat** [comment]
bbf17c9ecabe3e6f8782d1faf91c34ece361026b
2cc6640edffd96fcef16fe76901a1de7ab3326f0

**Pull request closed without merging**